src/query/scan_manager.c

```diff
@@ -2844,6 +2846,19 @@ scan_open_heap_scan (THREAD_ENTRY * thread_p, SCAN_ID * scan_id,
   hsidp->cache_recordinfo = cache_recordinfo;
   hsidp->recordinfo_regu_list = regu_list_recordinfo;
 
+  /* for scampling statistics. */
+  if (scan_type == S_HEAP_SAMPLING_SCAN && !is_partition_table)
+    {
+      int total_pages = 0;
+      if (file_get_num_total_user_pages (thread_p, cls_oid, &total_pages) != NO_ERROR)
```

Comment:
Calls the file_get_num_total_user_pages function for non-partition table cases.
However, the file_get_num_total_user_pages function executes the partition_get_partition_oids function to obtain the number of OIDs of the partition.

Can the function file_get_num_total_user_pages be used for something else?